upf/upf_cli.c

```diff
@@ -287,27 +285,28 @@ upf_nat_pool_add_del_command_fn (vlib_main_t * vm,
 	;
       else if (unformat (line_input, "block_size %u", &port_block_size))
 	;
-      else if (unformat (line_input, "vrf %u", &vrf_id))
-	;
-      else if (unformat (line_input, "name %_%v%_", &name))
+      // FIXME: This is a workaround to have 2 strings fetched from args
+      // There is kind of bug if two strings are present in command line
+      // and there are two appropriate unformat calls for it
+      // somehow it affects input line (wrong end-of-string indication or
+      // something like this). To avoid this, two strings could be picked
+      // in single unformat call. Probably Address Sanitaizer should be
+      // used to identify problems with unformatting input line
+      else if (unformat (line_input, "nwi %_%v%_ name %_%v%_", &nwi_s, &name))
```

Comment:
it seems that "%s" does not suffer from this problem. It would therefore be possible to scan the names as string and then convert them to vectors.
That would IMHO be a better workaround.

Reply:
Looking at the full function also makes me wonder if the fact that `name` is not initialized to zero might play a role in the problem.

Reply:
Initiation of name vectors to 0 resolves the issue, so I've removed W/A.
Thanks for good notice